src/views/Claims.vue

```diff
@@ -1,37 +1,109 @@
+                </b-row>
+            </template>
+        </b-container>
     </div>
 </template>
 
 <script lang="ts">
 import { Component, Vue } from 'vue-property-decorator';
 import { mapGetters, mapState } from 'vuex';
+import Web3 from 'web3';
+import { default as ABI_THX } from '../json/ABITHX.json';
+import { Contract } from 'web3-eth-contract';
 
 @Component({
     computed: { ...mapState('metamask', ['account', 'chainId']), ...mapGetters('metamask', ['isConnected']) },
 })
 export default class Claims extends Vue {
     account!: string;
-    chainId!: number;
+    web3!: Web3;
+    contract!: Contract;
+    reward!: number;
+    tokenAndAmount: Token[] = [];
 
+    /**
+     * Connects user to metamask, after that the reward variable is updated
+     */
     async connect() {
-        this.$store.dispatch('metamask/connect');
+        // set address of smart-contract, found in modules-solidity after command npx hardhat node
+        const address = '0x5E0A87862f9175493Cc1d02199ad18Eff87Eb400';
+        await this.$store.dispatch('metamask/connect');
+        this.contract = new this.web3.eth.Contract(ABI_THX as any, address);
+        // when connected trough metamask update reward variable
+        this.updateReward();
     }
-    async switchNetwork(networkId: number) {
-        this.$store.dispatch('metamask/requestSwitchNetwork', networkId);
+
+    /**
+     * Update the reward variable and get all unique tokens with their amount and stores it in the tokenAndAmount Object array
+     */
+    async updateReward() {
+        const positionAddress = 0;
+        const positonAmount = 2;
+        let _amount!: any;
+        this.reward = 0;
+        this.tokenAndAmount = [];
+        try {
+            const response = await this.contract.methods.getRewards().call();
+            // loop to set all unique tokens to the tokenAndAmount-array with their address and amount
+            for (let i = 0; i < response.length; i++) {
+                // cast to Number, because response returns hexadecimal
+                _amount = Number(response[i][positonAmount]._hex.toString());
+                this.reward += _amount;
+                // add all unique tokens to the tokenAndAmount-array including the amount of that specific token
+                this.tokenAndAmount.push({ token: response[i][positionAddress].toString(), amount: _amount });
+            }
+        } catch (err) {
+            console.log('Error: ' + err);
+        }
     }
 
-    get networkExpected() {
-        return this.chainId === 31337 ? 1 : 31337;
+    async payAllRewards() {
+        await this.contract.methods.withdrawBulk().call();
     }
 
-    created() {
-        this.$store.dispatch('metamask/checkPreviouslyConnected');
+    /**
+     * Transfers the reward of one token
+     * @param {string} address - The address of the token
+     */
+    async payOneReward(address: string) {
+        await this.contract.methods.withdraw(address).call();
```

Comment:
@wooyh2000 This one also seems more like a send instead of a call operation. The difference is basically read vs. write.